### ISSUE TITLE
Error messages during compilation: all Notices, Warnings and Errors always indicate Line 1 Column 0 no matter which actual line/column

### ISSUE THREAD

**Issue by JJFlash-IT** [neilsf/xc-basic3#185]
XC=Basic version 3.0.7

This doesn't always happen, but some specific cases generate this problem.
I can't seem to figure out what specifically triggers it, all I can offer to replicate the problem is this little program source code (it's the Superchase conversion I talked about in the FB group):
https://gitea.it/JJFlash/C64_Superchase/raw/branch/master/superchase-ported.xcbbas

It's full of undeclared variables, implicit conversions and GOTOs, on purpose (it's an almost 1:1 conversion of a very old TI99-4/A program taken from a book).

![errors-no-line](https://user-images.githubusercontent.com/48549431/174347520-50d71f35-b242-48d7-912c-ed0f000c34e0.png)
I broke the code on purpose to show than even Errors have this.

(As a side note: this little program seems to take too much time to compile. I also wrote a bigger and more complex program - a little virtual machine - and it takes less time. Is it maybe because of all the undeclared variables and implicit conversions?)

**Comment by JJFlash-IT** [comment]
**Of course** I had to find the specific cause of this *only after* posting this issue... :facepalm: :smile: 

It's the New Line delimiter: if the lines are separated only by LF, the compiler apparently doesn't count lines. If the same identical source file uses CRLF as separator, the compiler counts lines correctly.

*This seems to **not** happen when it comes to syntax errors: the correct line and column are indicated in LF-only files as well.*

Since XCB is available for other platforms too, not Windows only, I guess this should be considered to be fixed.

**Comment by neilsf** [comment]
Hi @JJFlash-IT 
Great catch, thanks! It'll be fixed soon.

**Comment by neilsf** [comment]
On what OS did you encounter this?
On Windows, XC=BASIC counts CRLF as newline. On Linux/Mac, it counts NL as newline. Maybe you wrote the BASIC source in Linux or Mac, and you're trying to compile it in Windows?

**Comment by JJFlash-IT** [comment]
I'm using Windows :) It happened because I downloaded an old BASIC program from archive.org, and then added the XCB code directly on it while commenting out the original code. That file used LF, not CRLF, as the line delimiter, and that's how I got into the problem.
So yeah, I suppose it's not a common scenario, especially since you've verified that on Linux and Mac this doesn't occur. But, it can happen :)
